# Convert all values to Booleans
Theses values are already in boolean format

- [1:8]
- [31:36]
- 42
- 118

In [1]:
# Imports

# Plots
%matplotlib inline
import mpld3
import matplotlib.pyplot as plt
mpld3.enable_notebook()

import pandas as pd
df = pd.read_csv('../dumps/16K.csv')
from toBoolean import *

## Feature 9 : the Checksum
10958 rows have value 0 <br>
=> 68% <br>
`0 if x == 0 else 1`

In [2]:
convert_f9(df['f9'])

0        1
1        1
2        0
3        0
4        0
        ..
16031    0
16032    1
16033    0
16034    1
16035    0
Name: f9, Length: 16036, dtype: int64

## Feature 10 : the Image Base
14 263 rows have value 4 194 304 <br>
=> 89% <br>
`1 if x == 4_194_304 else 0`

In [3]:
convert_f10(df['f10'])

0        1
1        1
2        1
3        1
4        1
        ..
16031    1
16032    1
16033    1
16034    1
16035    0
Name: f10, Length: 16036, dtype: int64

## Feature 11 : the Base of Code
13084 rows have value 4096 (82%) <br>
1022  rows have value 8192 (6%)

In [4]:
convert_f11(df['f11'])

0        1
1        1
2        2
3        0
4        2
        ..
16031    1
16032    1
16033    1
16034    1
16035    1
Name: f11, Length: 16036, dtype: int64

## Feature 12 : the OS Major version
10786 rows have value 4 (67%) <br>
3898 rows have value 5 (24%)

In [5]:
convert_f12(df['f12'])

0        1
1        2
2        1
3        1
4        1
        ..
16031    1
16032    1
16033    1
16034    1
16035    2
Name: f12, Length: 16036, dtype: int64

## Feature 13 : the OS Minor version
13235 rows have value 0 (83%)

In [6]:
convert_f13(df['f13'])

0        1
1        1
2        1
3        1
4        1
        ..
16031    1
16032    1
16033    1
16034    1
16035    0
Name: f13, Length: 16036, dtype: int64

## Feature 14 : the Size of Image
Random

In [7]:
convert_f14(df['f14'])

0        1
1        1
2        0
3        1
4        0
        ..
16031    1
16032    1
16033    0
16034    1
16035    0
Name: f14, Length: 16036, dtype: int64

## Feature 15 : the Size of Code

In [8]:
convert_f15(df['f15'])

0        1
1        1
2        0
3        0
4        0
        ..
16031    1
16032    0
16033    0
16034    0
16035    0
Name: f15, Length: 16036, dtype: int64

## Feature 16 : the Headers
6777 rows have value 1024 (42%) <br>
5502 rows have value 4096 (34%) <br>
2577 rows have value 512 (16%) <br>
1135 rows have value 1536 (7%)

In [9]:
convert_f16(df['f16'])

0        2
1        1
2        2
3        1
4        2
        ..
16031    1
16032    1
16033    2
16034    1
16035    4
Name: f16, Length: 16036, dtype: int64

## Feature 17 : the Size Of InitializedData

In [10]:
with pd.option_context('display.max_rows', None, 'display.max_columns', None):
    print(Feature 20 : the Size of Stack Commit)

SyntaxError: invalid syntax (<ipython-input-10-2675bf4da70d>, line 2)

# Feature 18 : the Size Of UninitializedData
13446 rows have value 0 (84%)<br>

In [12]:
convert_f18(df['f18'])

0        1
1        1
2        1
3        0
4        1
        ..
16031    1
16032    1
16033    1
16034    1
16035    1
Name: f18, Length: 16036, dtype: int64

## Feature 19 : the Size Of StackReserve
13520 rows have value 1048576 (84%)

In [14]:
convert_f19(df['f19'])

0        1
1        1
2        1
3        1
4        1
        ..
16031    1
16032    1
16033    1
16034    1
16035    0
Name: f19, Length: 16036, dtype: int64

## Feature 20 : the Size of Stack Commit
13948 rows have value 4096 (87%) <br>
1570 rows have value 16384 (10%) <br>
229 rows have value 8192 (1%) <br>
93 rows have value 65536 (0.6%)

In [15]:
convert_f20(df['f20'])

0        1
1        1
2        1
3        1
4        1
        ..
16031    1
16032    1
16033    1
16034    1
16035    4
Name: f20, Length: 16036, dtype: int64

## Feature 21 : the Section Alignment
15005 rows have value 4096 (94%) <br>
996 rows have value 8192 (6%)

In [16]:
convert_f21(df['f21'])

0        1
1        1
2        2
3        1
4        2
        ..
16031    1
16032    1
16033    1
16034    1
16035    1
Name: f21, Length: 16036, dtype: int64

## Feature 22 : the number of standards sections the PE holds
4170 rows have value 4 (26%) <br>
3271 rows have value 3 (20%) <br>
2675 rows have value 2 (17%) <br>
2200 rows have value 5 (14%) <br>
1703 rows have value 1 (11%) <br>
... <br>
` x if x < 9 else 9`

In [17]:
convert_f22(df['f22'])

0        3.0
1        5.0
2        3.0
3        1.0
4        3.0
        ... 
16031    4.0
16032    2.0
16033    4.0
16034    2.0
16035    3.0
Name: f22, Length: 16036, dtype: float64

## Feature 23 : the number of non-standards sections the PE holds
10272 rows have value 0 (64%) <br>
1939 rows have value 2 (12%) <br>
1571 rows have value 1 (10%) <br>
1557 rows have value 3 (10%) <br>
` x if x < 4 else 4`

In [18]:
convert_23(df['f23'])

0.0     10272
2.0      1939
1.0      1571
3.0      1557
4.0       322
5.0       174
8.0        81
6.0        54
9.0        27
10.0        8
12.0        7
11.0        6
7.0         4
14.0        2
18.0        2
28.0        1
39.0        1
50.0        1
61.0        1
21.0        1
36.0        1
13.0        1
37.0        1
46.0        1
47.0        1
Name: f23, dtype: int64

## Feature 24 : the ratio between the number of standards sections found and the number of all sections found in the PE under analysis
10272 rows have value 1 (64%)

In [19]:
convert_f24(df['f24'])

1.000    10272
0.333     1067
0.000      805
0.625      700
0.800      678
         ...  
0.944        1
0.368        1
0.080        1
0.067        1
0.027        1
Name: f24, Length: 61, dtype: int64

## Feature 25 : the number of Executable sections the PE holds
9420 rows have value 1 (59%) <br>
4743 rows have value 2 (30%) <br>
1113 rows have value 0 (7%)

In [20]:
convert_f25(df['f25'])

1.0     9420
2.0     4743
0.0     1113
3.0      333
4.0      272
6.0       98
5.0       15
7.0       12
8.0        9
11.0       4
9.0        2
14.0       2
10.0       2
12.0       2
39.0       1
38.0       1
50.0       1
46.0       1
62.0       1
13.0       1
22.0       1
37.0       1
48.0       1
Name: f25, dtype: int64

## Feature 26 : the number of Writable sections the PE holds